Core/Persistence/Legacy/Content/FieldValue/Converter/EnhancedSelectionConverter.php

```diff
@@ -63,6 +119,39 @@ public function toStorageFieldDefinition( FieldDefinition $fieldDef, StorageFiel
      */
     public function toFieldDefinition( StorageFieldDefinition $storageDef, FieldDefinition $fieldDef )
     {
+        $simpleXml = simplexml_load_string( $storageDef->dataText5 );
+        $options = array();
+        $isMultiple = false;
+        $delimiter = false;
+        $query = false;
+
+        if ( $simpleXml !== false )
+        {
+            foreach ( $simpleXml->options->option as $option )
+            {
+                $options[(int)$option["id"]] = array( "id" => (string)$option["id"],
```

Comment:
Also, is there any specific reason why you set the array keys to `id`? Why not just use `$options[] = array( ... )`? I'm worried that someone somewhere will try to use this array as zero-based, while in reality it is be one-based.